packages/quantic/force-app/main/default/lwc/quanticColoredResultBadge/quanticColoredResultBadge.js

```diff
@@ -70,27 +94,29 @@ export default class QuanticColoredResultBadge extends LightningElement {
 
   /**
    * Generates the secondary color.
-   * @param {number} lightnessAmount the amount of lightness with which to update the primary color.
    * @returns {string}
    */
-  generateSecondaryColor(lightnessAmount) {
+  generateSecondaryColor() {
     const {r, g, b} = HEXToRGB(this.color);
 
     if (invalidRGBValues(r, g, b)) {
       this.invalidColor = true;
       return defaultSecondaryColor;
     }
 
-    const HSLColor = RGBToHSL(r, g, b);
-    const {h, s} = HSLColor;
-    let l = HSLColor.l;
+    const luminance =
+      (r / 255.0) * 0.2126 +
+      (g / 255.0) * 0.7152 +
+      (b / 255.0) * 0.0722
 
-    if (l >= 50) {
-      l -= lightnessAmount;
-    } else {
-      l += lightnessAmount;
-    }
+    return luminance >= 0.50 ? 'black' : 'white'
```

Comment:
This could be worth a comment to explain the logic here. So in 6 months we know what those values are or where they are from.

Also I think that if you removed the usage of `RGBToHSL` it might not be in use anywhere else? In that case I would remove it.

Reply:
J'ai pas vraiment d'explications. Je sais juste que c'est dérivé de l'algo du backend du builder...

Reply:
Cf. le ticket: https://coveord.atlassian.net/browse/SVCC-2706